presto-main/src/test/java/com/facebook/presto/server/TestQuerySessionSupplier.java

```diff
@@ -95,6 +100,50 @@ public void testCreateSession()
     }
 
     @Test
+    public void testApplySessionPropertyConfigurationManager()
+            throws Exception
+    {
+        HttpServletRequest request = new MockHttpServletRequest(
+                ImmutableListMultimap.<String, String>builder()
+                        .put(PRESTO_USER, "testUser")
+                        .put(PRESTO_SOURCE, "testSource")
+                        .put(PRESTO_CATALOG, "testCatalog")
+                        .put(PRESTO_SCHEMA, "testSchema")
+                        .put(PRESTO_LANGUAGE, "zh-TW")
+                        .put(PRESTO_TIME_ZONE, "Asia/Taipei")
+                        .put(PRESTO_CLIENT_INFO, "client-info")
+                        .put(PRESTO_CLIENT_TAGS, "tag1,tag2 ,tag3, tag2")
+                        .put(PRESTO_SESSION, QUERY_MAX_MEMORY + "=1GB")
+                        .put(PRESTO_SESSION, DISTRIBUTED_JOIN + "=true," + HASH_PARTITION_COUNT + " = 43")
+                        .put(PRESTO_PREPARED_STATEMENT, "query1=select * from foo,query2=select * from bar")
+                        .build(),
+                "testRemote");
+
+        HttpRequestSessionContext context = new HttpRequestSessionContext(request);
+        QuerySessionSupplier sessionSupplier = new QuerySessionSupplier(
+                createTestTransactionManager(),
+                new AllowAllAccessControl(),
+                new SessionPropertyManager());
+        List<AlwaysMatchSessionPropertyRule> rules = ImmutableList.of(
+                new AlwaysMatchSessionPropertyRule(1, ImmutableMap.of("testCatalog.key1", "1", "key2", "2", "key3", "3")),
+                new AlwaysMatchSessionPropertyRule(100, ImmutableMap.of("key2", "20")),
+                new AlwaysMatchSessionPropertyRule(10, ImmutableMap.of(QUERY_MAX_MEMORY, "10GB", "testCatalog.key1", "10")));
+
+        SessionPropertyConfigurationManagerFactory factory = new TestingSessionPropertyConfigurationManagerFactory(rules);
+        sessionSupplier.addSessionPropertyConfigurationManager(factory);
+        sessionSupplier.setSessionPropertyConfigurationManager(factory.getName(), ImmutableMap.of());
```

Comment:
This is because it's a test. In reality, `add...` is called in `PluginManager` to register all available plugins, and `set...` is called here in this class according to config files (hence `@VisibleForTesting`.